### PR TITLE
chore(hybridcloud) Reduce outbox generation for orgauthtoken

### PR DESCRIPTION
**Pull request by markstory**
We currently emit two outbox messages for every request that uses an OrgAuthToken. Often the usage patterns for org auth tokens are many requests close together in time. By only emitting one outbox per unique project set per minute we should be able to lower the number of redundant outboxes that are created without greatly impacting the accuracy of the last used time and project ids.

Refs HC-1191

### CODE REVIEW

File: src/sentry/models/orgauthtoken.py [hunk 1]
```diff
@@ -151,10 +152,21 @@ def get_org_auth_token_id_from_auth(auth: object) -> int | None:
...
     org_auth_token_id = get_org_auth_token_id_from_auth(auth)
     organization_id = getattr(auth, "organization_id", None)
-    if org_auth_token_id is not None and organization_id is not None:
-        orgauthtoken_service.update_orgauthtoken(
-            organization_id=organization_id,
-            org_auth_token_id=org_auth_token_id,
-            date_last_used=timezone.now(),
-            project_last_used_id=project_ids[0] if len(project_ids) > 0 else None,
-        )
+    if not org_auth_token_id and not organization_id:
```
Comment (szokeasaurusrex) [inline src/sentry/models/orgauthtoken.py:155]: Why are we no longer checking these values explicitly against `None`, like we are in the previous version? The behavior here could be different if either value is false-y but not `None`; is this behavior change intentional?

File: src/sentry/models/orgauthtoken.py [hunk 2]
```diff
@@ -151,10 +152,21 @@ def get_org_auth_token_id_from_auth(auth: object) -> int | None:
...
+        return
+
+    # Debounce updates, as we often get bursts of requests when customer
+    # run CI or deploys and we don't need second level precision here.
+    # We vary on the project ids so that unique requests still make updates
+    project_segment = ",".join([str(i) for i in project_ids])
```
Comment (szokeasaurusrex) [inline src/sentry/models/orgauthtoken.py:161]:
```suggestion
    project_segment = ",".join(str(i) for i in project_ids)
```

`join` takes an iteratable, so we don't need to build an array here

File: tests/sentry/api/endpoints/test_organization_releases.py [hunk 3]
```diff
@@ -1782,7 +1782,7 @@ def test_org_auth_token(self):
             good_token_str = generate_token(org.slug, "")
             OrgAuthToken.objects.create(
                 organization_id=org.id,
-                name="token 1",
+                name="token 2",
                 token_hashed=hash_token(good_token_str),
                 token_last_characters="ABCD",
                 scope_list=["org:ci"],
```
Comment (szokeasaurusrex) [inline tests/sentry/api/endpoints/test_organization_releases.py:1788]: Why are these changes needed? And, should we perhaps add a test for the caching functionality?

**Review by szokeasaurusrex — COMMENTED** [review]
Requested some minor changes and left some questions, but overall, I think limiting the updates to once per minute seems like a good idea.

File: tests/sentry/api/endpoints/test_organization_releases.py (hunk 3 above)
Comment (markstory) [reply tests/sentry/api/endpoints/test_organization_releases.py:1788]: There were two tokens with a name of 'token 1' for the same org which made debugging harder.

File: src/sentry/models/orgauthtoken.py (hunk 1 above)
Comment (markstory) [reply src/sentry/models/orgauthtoken.py:155]: Projects and orgs cannot have id values of 0. I'll put the none checks back though.

**Review by szokeasaurusrex — APPROVED** [review]
Looks good to me!
